src/components/links/__tests__/Link-test.tsx

```diff
@@ -0,0 +1,138 @@
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+import { screen } from '@testing-library/react';
+import { MemoryRouter, Route, Routes, useLocation } from 'react-router-dom';
+import { LinkSubdued } from '..';
+import { render } from '../../../common/helpers/testUtils';
+import { Link } from '../Link';
+
+beforeAll(() => {
+  const { location } = window;
+  delete (window as { location?: Location }).location;
+  window.location = { ...location, href: '' };
+});
+
+beforeEach(() => {
+  jest.clearAllMocks();
+});
+
+// This functionality won't be needed once we update the breadcrumbs
+it('should remove focus after link is clicked', async () => {
+  const { user } = setupWithMemoryRouter(
+    <Link icon={<div>Icon</div>} shouldBlurAfterClick to="/initial">
+      Test
+    </Link>,
+  );
+
+  await user.click(screen.getByRole('link'));
+
+  expect(screen.getByRole('link')).not.toHaveFocus();
+});
+
+it('should prevent default when preventDefault is true', async () => {
+  const { user } = setupWithMemoryRouter(
+    <Link shouldPreventDefault to="/second">
+      Test
+    </Link>,
+  );
+
+  expect(screen.getByText('/initial')).toBeVisible();
+
+  await user.click(screen.getByRole('link'));
+
+  // prevent default behavior of page navigation
+  expect(screen.getByText('/initial')).toBeVisible();
+  expect(screen.queryByText('/second')).not.toBeInTheDocument();
+});
+
+it('should stop propagation when stopPropagation is true', async () => {
+  const buttonOnClick = jest.fn();
+
+  const { user } = setupWithMemoryRouter(
+    <button onClick={buttonOnClick} type="button">
+      <Link shouldStopPropagation to="/second">
+        Test
+      </Link>
+    </button>,
+  );
+
+  await user.click(screen.getByRole('link'));
+
+  expect(buttonOnClick).not.toHaveBeenCalled();
+});
+
+it('should call onClick when one is passed', async () => {
+  const onClick = jest.fn();
+  const { user } = setupWithMemoryRouter(
+    <Link onClick={onClick} shouldStopPropagation to="/second">
+      Test
+    </Link>,
+  );
+
+  await user.click(screen.getByRole('link'));
+
+  expect(onClick).toHaveBeenCalled();
+});
+
+it('internal link should be clickable', async () => {
+  const { user } = setupWithMemoryRouter(<Link to="/second">internal link</Link>);
+  expect(screen.getByRole('link')).toBeVisible();
+
+  await user.click(screen.getByRole('link'));
+
+  expect(screen.getByText('/second')).toBeVisible();
+});
+
+it('external links are indicated by OpenNewTabIcon', () => {
+  setupWithMemoryRouter(<Link to="https://google.com">external link</Link>);
+  expect(screen.getByRole('link')).toBeVisible();
+
+  expect(screen.getByRole('img', { hidden: true })).toBeInTheDocument();
```

Comment:
We can remove the `{hidden: true}` and use the toBeVisible once the accessibility is fixed.